backend/dataall/modules/s3_datasets_shares/services/share_managers/s3_access_point_share_manager.py

```diff
@@ -568,33 +575,31 @@ def update_dataset_bucket_key_policy(self):
                 self.generate_enable_pivot_role_permissions_policy_statement(pivot_role_name, self.dataset_account_id)
             )
 
-            if DATAALL_ACCESS_POINT_KMS_DECRYPT_SID in statements.keys():
-                logger.info(
-                    f'KMS key policy contains share statement {DATAALL_ACCESS_POINT_KMS_DECRYPT_SID}, '
-                    f'updating the current one'
-                )
-                statements[DATAALL_ACCESS_POINT_KMS_DECRYPT_SID] = self.add_target_arn_to_statement_principal(
-                    statements[DATAALL_ACCESS_POINT_KMS_DECRYPT_SID], target_requester_arn
-                )
-            else:
-                logger.info(
-                    f'KMS key does not contain share statement {DATAALL_ACCESS_POINT_KMS_DECRYPT_SID}, '
-                    f'generating a new one'
-                )
-                statements[DATAALL_ACCESS_POINT_KMS_DECRYPT_SID] = self.generate_default_kms_decrypt_policy_statement(
-                    target_requester_arn
-                )
+            for target_sid in perms_to_sids(self.share.permissions, SidType.KmsAccessPointPolicy):
```

Comment:
do we not have to add logic for revoke (at `def delete_dataset_bucket_key_policy()`) similar to how we do for bucket share?

Reply:
good catch, will push fixes